spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/AddressProviderConfiguration.java

```
@@ -0,0 +1,76 @@
+
+    private static final Logger log = LoggerFactory.getLogger(lookup().lookupClass());
+
+    @Autowired
+    private Environment environment;
+
+    @Bean
+    MicroserviceAddressProvider microserviceAddressProvider() {
+        String microserviceHost = environment.getProperty("microservice.host", resolveMicroserviceLocalhost());
+        Integer microservicePort = Integer.valueOf(environment.getProperty("server.port", "8080"));
+        return new MicroserviceAddressProvider(microserviceHost, microservicePort);
+    }
+
+    public static String resolveMicroserviceLocalhost() {
+        return getIpAddress();
+    }
+
+    /**
+     * Return a non loopback IPv4 address for the machine running this process.
+     * If the machine has multiple network interfaces, the IP address for the
+     * first interface returned by {@link java.net.NetworkInterface#getNetworkInterfaces}
+     * is returned.
+     *
+     * @return non loopback IPv4 address for the machine running this process
+     *
+     * @see java.net.NetworkInterface#getNetworkInterfaces
+     * @see java.net.NetworkInterface#getInetAddresses
+     */
+    public static String getIpAddress() {
+        try {
+            for(Enumeration<NetworkInterface> enumNic = NetworkInterface.getNetworkInterfaces();
+                enumNic.hasMoreElements();) {
+                NetworkInterface ifc = enumNic.nextElement();
+                if (ifc.isUp()) {
+                    for (Enumeration<InetAddress> enumAddr = ifc.getInetAddresses();
+                         enumAddr.hasMoreElements(); ) {
+                        InetAddress address = enumAddr.nextElement();
+                        if (address instanceof Inet4Address && !address.isLoopbackAddress()) {
+                            return address.getHostAddress();
+                        }
+                    }
+                }
+            }
+        }
+        catch (IOException e) {
+            // ignore
```

log an error?  I don't like to ignore exceptions much unless you can convince me otherwise.